sql/core/src/main/scala/org/apache/spark/sql/execution/python/ArrowEvalPythonExec.scala

```diff
@@ -51,10 +51,12 @@ case class ArrowEvalPythonExec(udfs: Seq[PythonUDF], output: Seq[Attribute], chi
       outputIterator.map(new ArrowPayload(_)), context)
 
     // Verify that the output schema is correct
-    val schemaOut = StructType.fromAttributes(output.drop(child.output.length).zipWithIndex
-      .map { case (attr, i) => attr.withName(s"_$i") })
-    assert(schemaOut.equals(outputRowIterator.schema),
-      s"Invalid schema from pandas_udf: expected $schemaOut, got ${outputRowIterator.schema}")
+    if (outputIterator.nonEmpty) {
```

Comment:
After `outputIterator` is consumed by `ArrowConverters`, can `nonEmpty` return a meaningful value?

Reply:
I guess we should use `outputRowIterator`.
Btw how about using `hasNext` instead of `nonEmpty`?

Reply:
ooops!  that was a typo and you're right, it should have been `outputRowIterator`